### PR TITLE
Release 2/sps

### PR DESCRIPTION
**Pull request by mike-gangl**
Draft PR for defining SounderSIPS based tests on WPS-T endpoints.

Missing:
* GetResults
* Deploy/Undeploy
* Cancel
* Non-nominal states (e.g. error handling)

@anilnatha and @rtapella i've added you as reviewers so you can start to see how the test cases will directly relate to the jupyter notebook work you need to do. This is not 'implementation spec' yet, we need buy-in from the developers before we can code to this, but i think these qualify as heuristics for the work to be done in the current form

### CODE REVIEW

**Review by LucaCinquini — CHANGES_REQUESTED** [review]
To be completely pedantic, I suggest the following change:

In the 2 scenarios "request L1A/L1B processing", the WPS-T response has status code 200 (not 302), and it contains a URL which the client can follow to monitor the job status (which is the last scenario in the document).

There are also a few misspellings:
- "retrievel"
- "Aglorithm"

**Review by LucaCinquini — APPROVED** [review]
Thanks for the changes. looks good.

**Comment by rtapella** [comment]
Aglorithm Ctlalalog

File: features/sps/process.feature [hunk 1]
```diff
@@ -1,11 +1,52 @@
...
+    And the OGC StatusInfo page returns a HTTP 200
+    And the OGC StatusInfo processing status is one of "Succeeded", "Failed", "Accepted", or "Running"
 
-  Scenario: Request Processing from an Algorithm deployment
+  Scenario: Request L1B Processing from an Algorithm deployment
+    Given the SoundsSIPS L1B algorithm has been deployed to the ADES
+    And SounderSIPS L1A data exists in the Unity system
+    When a WPS-T request is made to execute the job and the defined L1A Data
+    Then a WPS-T response is a 201
+    And the response include a Location header
```
Comment (rtapella) [inline features/sps/process.feature:43]: includes

File: features/sps/process.feature [hunk 2]
```diff
@@ -1,11 +1,52 @@
...
+    And the response page returns a HTTP 200
+    And the response page includes the one or more input element
+    # more to add here?
+
+  Scenario: Request L1A Processing from an Algorithm deployment
+    Given the SoundsSIPS L1A algorithm has been deployed to the ADES
+    And SounderSIPS L0 data exists for a 2 hour block of Data in the Unity system
+    When a WPS-T request is made to execute the job and the defined L0 Data
+    Then a WPS-T response is a 201
+    And the response include a Location header
```
Comment (rtapella) [inline features/sps/process.feature:33]: includes

File: features/sps/process.feature [hunk 3]
```diff
@@ -1,11 +1,52 @@
...
+    Given the SoundsSIPS L1A algorithm has been deployed to the ADES
+    When DescribeProcess is called on the WPS-T endpoint for the L1A Algorithm
+    Then the WPS-T endpoint responds with a ProcessOfferings response
+    And the response page returns a HTTP 200
+    And the response page includes the one or more input element
+    # more to add here?
+
+  Scenario: Request L1A Processing from an Algorithm deployment
+    Given the SoundsSIPS L1A algorithm has been deployed to the ADES
+    And SounderSIPS L0 data exists for a 2 hour block of Data in the Unity system
```
Comment (rtapella) [inline features/sps/process.feature:30]: Presumably this embeds some knowledge of the L1A algorithm... is there no way to query SPS/WPS-T directly to say "are all the requirements met for this algorithm to run"?

File: features/sps/process.feature [hunk 4]
```diff
@@ -1,11 +1,52 @@
...
+    When a WPS-T request is made to execute the job and the defined L0 Data
+    Then a WPS-T response is a 201
+    And the response include a Location header
+    And the Location header directs users to an OGC StatusInfo page
```
Comment (rtapella) [inline features/sps/process.feature:34]: "page" to me implies a GUI... in this case does it mean a RESTful URL?

**Review by rtapella — APPROVED** [review]

File: features/sps/process.feature (hunk 4 above)
Comment (mike-gangl) [reply features/sps/process.feature:34]: good call, i changed it to response and documents (which kind of infers 'page' but not technically)

File: features/sps/process.feature (hunk 3 above)
Comment (mike-gangl) [reply features/sps/process.feature:30]: this is a good thread to pull on, and i think its important to think of generic testing in the future, but we're not testing all of the input/output variants of the WPS here, we're testing the major interfaces. Unit/integration testing on the WPS-T deployment should cover some generic testing, and the projects wil need to bring their own validation at some point. there is a fuzzy area where we could write some validation, and that's probably valuable from a user sanity perspective, just not right now.

**Pull request merged**